search-app/src/cmr/search/services/acls/collections_cache.clj

```diff
@@ -16,16 +23,39 @@
 (defn create-cache
   "Creates a new empty collections cache."
   []
-  (mem-cache/create-in-memory-cache))
+  ;;(mem-cache/create-in-memory-cache)
+  (redis-cache/create-redis-cache {:keys-to-track [:collections-for-gran-acls] :ttl (* 15 60)}))
 
-(def cache-key
-  :collections-for-gran-acls)
+(defn- clj-times->time-strs
+  "Take a map and convert any date objects into strings so the map can be cached.
+   This can be reversed with time-strs->clj-times."
+  [data]
+  (walk/postwalk
+   #(if (true? (instance? org.joda.time.DateTime %))
+      (cmr.common.date-time-parser/clj-time->date-time-str %)
+      %)
+   data))
+
+(defn- time-strs->clj-times
+  "Take a map which has string dates and convert them to DateTime objects. This
+   can be reversed with clj-times->time-strs."
+  [data]
+  (walk/postwalk
+   #(if (and (instance? String %) (cmr.common.date-time-parser/try-parse-datetime %))
```

Comment:
I think (if-let [parsed-date (and (instance? String %) (parser/try-parse-datetime %))]
 would be better then you don't have to parse the date twice.  

If this doesn't work then use can you use a let to parse the date once?

Reply:
got it to work and wrote a test for this function